src/WebAuthn/Signature.hs

```diff
@@ -41,6 +42,12 @@ verifySig (PubRSA pub) sig dat
   | Just dat' <- parseRS256Signature (RSA.ep pub sig), dat' == BA.convert (hashWith SHA256 dat) = pure ()
   | otherwise = Left $ SignatureFailure "RS256"
 
+hasMatchingAlg :: PublicKey -> PubKeyCredAlg -> Bool
+hasMatchingAlg key algo =
+    case key of
+      PubEC (EC.PublicKey curve _) -> algo == ES256 && (curve == (EC.getCurveByName EC.SEC_p256r1))
```

Comment:
Nitpick: `(curve == (EC.getCurveByName EC.SEC_p256r1))` can just be `curve == EC.getCurveByName EC.SEC_p256r1`